pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -95,170 +92,57 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
-					"VMSelector", vmSelector.(*cloudv1alpha1.VirtualMachineSelector))
-			}
-		}
-
-		for i := range selector.Spec.VMSelector {
-			if err := accPoller.vmSelector.Add(&selector.Spec.VMSelector[i]); err != nil {
-				r.Log.Error(err, "unable to add selector into indexer",
-					"VMSelector", selector.Spec.VMSelector[i])
-			}
-		}
-
-		vmList := &cloudv1alpha1.VirtualMachineList{}
-		if err := r.List(context.TODO(), vmList, client.MatchingFields{
-			virtualMachineIndexerByCloudAccount: selector.Name}, client.InNamespace(selector.Namespace)); err != nil {
-			r.Log.Error(err, "unable to get virtual machines for external entity selector",
-				"Name", selector.Name)
-		} else {
-			var vms []*cloudv1alpha1.VirtualMachine
-			for i := range vmList.Items {
-				vm := &vmList.Items[i]
-				// vm selector changed, trigger to recompute.
-				vms = append(vms, vm)
-			}
-			if err := accPoller.doVirtualMachineOperations(vms); err != nil {
-				r.Log.Error(err, "unable to update virtual machines")
-			}
-		}
+	accountNamespacedName := &types.NamespacedName{
+		Namespace: selector.Namespace,
+		Name:      selector.Spec.AccountName,
 	}
-
-	cloudInterface, err := cloudprovider.GetCloudInterface(common.ProviderType(accPoller.cloudType))
+	r.selectorToAccountMap[*selectorNamespacedName] = *accountNamespacedName
+	err, accPoller := r.Poller.updateAccountPoller(accountNamespacedName, selector)
 	if err != nil {
-		if !preExists {
-			_ = r.processDelete(selectorNamespacedName)
-		}
 		return err
 	}
 
-	err = cloudInterface.AddAccountResourceSelector(accPoller.namespacedName, selector)
+	cloudInterface, err := cloudprovider.GetCloudInterface(common.ProviderType(accPoller.cloudType))
 	if err != nil {
-		if !preExists {
-			_ = r.processDelete(selectorNamespacedName)
-		}
-		r.Log.Info("selector add failed", "selector", selectorNamespacedName,
-			"poller-exists", preExists)
+		_ = r.processDelete(selectorNamespacedName)
 		return err
 	}
 
-	if !preExists {
-		go wait.Until(accPoller.doAccountPoller, time.Duration(accPoller.pollIntvInSeconds)*time.Second, accPoller.ch)
-	}
-	return nil
-}
-
-func (r *CloudEntitySelectorReconciler) processDelete(selectorNamespacedName *types.NamespacedName) error {
-	poller := r.removeAccountPoller(selectorNamespacedName)
-	if poller == nil {
-		return nil
-	}
-	cloudInterface, err := cloudprovider.GetCloudInterface(common.ProviderType(poller.cloudType))
+	err = cloudInterface.AddAccountResourceSelector(accPoller.namespacedName, selector)
 	if err != nil {
+		_ = r.processDelete(selectorNamespacedName)
+		r.Log.Info("selector add failed", "selector", selectorNamespacedName)
 		return err
 	}
-	cloudInterface.RemoveAccountResourcesSelector(poller.namespacedName, selectorNamespacedName.Name)
+
+	// Stop and start goroutine in order to trigger account poller soon after CES add/update.
+	close(accPoller.ch)
+	accPoller.ch = make(chan struct{})
+	r.Log.Info("Restarting poller in CES")
```

Comment:
To keep the logging consistent within a file/package.
```suggestion
	r.Log.Info("restarting poller in CES")
```

Reply:
restarting account poller on CES update. Consider making this a function in account poller file.

Reply:
Added, didn't add earlier because there is no other consumer, probably you want to keep all operations on accountpoller in one file. Hence, added it.